pool/hub.go

```diff
@@ -142,7 +141,7 @@ type HubConfig struct {
 type Hub struct {
 	clients int32 // update atomically.
 
-	db             *bolt.DB
+	db             Database
```

Comment:
Just noticed a the hub config and the hub all have references to the db, likely oversight on my part. I think we should remove one however: the hub reference I'm thinking.

Reply:
Personally I think removing the one from config makes more sense. The database is one dependency, the config is another. Each is a saparate thing with distinct responsibilities - the config holds simple values which define how the hub should behave, the database is a persistent storage device.

Reply:
On second thought, I wont make that change because that will conflict with how the rest of the dcrpool components are setup. They all have the database stored in their config, so I will maintain that for hub as well.